File: src/LearningApp.Service/LearningApp.Service.API/Controllers/AuthorizationController.cs
using LearningApp.Service.API.Contracts.Authorization;
using LearningApp.Service.API.Contracts.Authorization.Requests;
using LearningApp.Service.API.Contracts.Authorization.Responses;
using LearningApp.Service.API.Managers;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.Logging;

namespace LearningApp.Service.API.Controllers
{
	[Route("[controller]")]
	[ApiController]
	public class AuthorizationController : ApiControllerBase, IAuthorizationController
	{
		private readonly IAuthorizationManager _authorizationManager;

		public AuthorizationController(ILogger<AuthorizationController> logger, IAuthorizationManager authorizationManager) : base(logger)
		{
			_authorizationManager = authorizationManager;
		}

		/// <summary>
		/// Авторизоваться по почте и паролю.
		/// </summary>
		/// <param name="loginRequest"></param>
		/// <returns></returns>
		/// <response code="200">Информация об access токене</response>
		/// <response code="400">Указаны некорректные данные</response>
		/// <response code="401">Неверный пароль</response>
		/// <response code="404">Пользователь с указанной почтой не найден</response>
		[HttpPost("login")]
		[ProducesResponseType<AccessTokenResponse>(StatusCodes.Status200OK)]
		[ProducesResponseType(typeof(void), StatusCodes.Status400BadRequest)]
		[ProducesResponseType(typeof(void), StatusCodes.Status401Unauthorized)]
		[ProducesResponseType(typeof(void), StatusCodes.Status404NotFound)]
		public IActionResult Login([FromBody] LoginRequest loginRequest)
		{
			return _authorizationManager.TryLogin(loginRequest).ToActionResult(loginRequest?.Language ?? CurrentUserLanguage);
		}

		/// <summary>
		/// Зарегистрироваться по почте, имени пользователя, паролю и языку.
		/// </summary>
		/// <param name="registrationRequest"></param>
		/// <returns></returns>
		/// <response code="200">Информация об access токене</response>
		/// <response code="400">Указаны некорректные данные</response>
		/// <response code="409">Пользователь с указанными данными уже зарегистрирован</response>
		[HttpPost("register")]
		[ProducesResponseType<AccessTokenResponse>(StatusCodes.Status200OK)]
		[ProducesResponseType(typeof(void), StatusCodes.Status400BadRequest)]
		[ProducesResponseType(typeof(void), StatusCodes.Status409Conflict)]
		public IActionResult Register([FromBody] RegistrationRequest registrationRequest)
		{
			return _authorizationManager.TryRegister(registrationRequest).ToActionResult(registrationRequest?.Language ?? CurrentUserLanguage);
		}

		/// <summary>
		/// Обновить access токен сессии по refresh токену.
		/// </summary>
		/// <returns></returns>
		/// <response code="401">Сессия не валидна</response>
		/// <response code="403">Некорректные параметры сессии</response>
		[Authorize]
		[HttpPost("refresh")]
		[ProducesResponseType<AccessTokenResponse>(StatusCodes.Status200OK)]
		[ProducesResponseType(typeof(void), StatusCodes.Status401Unauthorized)]
		[ProducesResponseType(typeof(void), StatusCodes.Status403Forbidden)]
		public IActionResult Refresh()
		{
			return _authorizationManager.TryRefresh(User).ToActionResult(CurrentUserLanguage);
		}
	}
}

File: src/LearningApp.Android/app/src/main/java/com/example/learningappandroid/api/controllers/AuthorizationController.kt
package com.example.learningappandroid.api.controllers

import com.example.learningappandroid.api.contracts.authorization.IAuthorizationController
import com.example.learningappandroid.api.contracts.authorization.requests.LoginRequest
import com.example.learningappandroid.api.contracts.authorization.requests.RegistrationRequest
import com.example.learningappandroid.api.contracts.authorization.responses.AccessTokenResponse
import retrofit2.Response
import retrofit2.Retrofit
import retrofit2.http.Body
import retrofit2.http.Header

class AuthorizationController(retrofit: Retrofit) : IAuthorizationController {

	private val api: IAuthorizationController = retrofit.create(IAuthorizationController::class.java)

	override suspend fun login(@Body loginRequest: LoginRequest): Response<AccessTokenResponse> {
		return api.login(loginRequest)
	}

	override suspend fun register(@Body registrationRequest: RegistrationRequest): Response<AccessTokenResponse> {
		return api.register(registrationRequest)
	}

	override suspend fun refresh(@Header("Authorization") bearerToken: String): Response<AccessTokenResponse> {
		return api.refresh(bearerToken)
	}

	override suspend fun logout(@Header("Authorization") bearerToken: String): Response<Unit> {
		return api.logout(bearerToken)
	}
}